app/services/metrics/metrics_processor.rb

```diff
@@ -0,0 +1,73 @@
+module Metrics
+  ##
+  # Does the processing of the Github events previously stored and leaves the
+  # generated metrics in the Metrics table.
+  #
+  # It takes care of processing only the metrics that are pending to be
+  # processed.
+  class MetricsProcessor < BaseService
+    ##
+    # Processes the metrics
+    def call
+      process_all_metrics
+    end
+
+    private
+
+    ##
+    # Returns all the defined metrics to process.
+    def all_metrics_definitions
+      MetricsDefinition.all
+    end
+
+    ##
+    # Returns the concrete MetricsProcessor to process the metrics definition.
+    def metrics_processor_for(metrics_definition)
+      metrics_definition.metrics_processor.constantize
+    end
+
+    ##
+    # Makes each metric defined to process all the events.
+    def process_all_metrics
+      metrics_definitions = all_metrics_definitions
+
+      events = events_to_process_for(metrics_definitions)
```

Comment:
shouldn't you grab the events to process in a metrics definition basis instead of doing that "min" processed at event logic?